File: qasdk/src/main/java/co/astrnt/qasdk/repository/BaseRepository.java
package co.astrnt.qasdk.repository;

import android.content.Context;

import co.astrnt.qasdk.AstrntSDK;
import co.astrnt.qasdk.core.AstronautApi;
import co.astrnt.qasdk.utils.ServiceUtils;
import co.astrnt.qasdk.utils.services.SendLogService;

public class BaseRepository {

    protected final AstronautApi mAstronautApi;
    protected AstrntSDK astrntSDK = new AstrntSDK();

    public BaseRepository(AstronautApi astronautApi) {
        mAstronautApi = astronautApi;
    }

    public void sendLog(Context context) {
        if (!ServiceUtils.isMyServiceRunning(context, SendLogService.class)) {
            SendLogService.start(context);
        }
    }

}

File: qasdk/src/main/java/co/astrnt/qasdk/repository/BaseRepository.kt
package co.astrnt.qasdk.repository

import android.content.Context
import androidx.work.Constraints
import androidx.work.NetworkType
import androidx.work.OneTimeWorkRequest
import androidx.work.WorkManager
import co.astrnt.qasdk.AstrntSDK
import co.astrnt.qasdk.core.AstronautApi
import co.astrnt.qasdk.utils.services.SendLogWorkerJava


open class BaseRepository(@JvmField val mAstronautApi: AstronautApi) {
    @JvmField var astrntSDK = AstrntSDK()
    fun sendLog(context: Context?) {
        val constraints = Constraints.Builder()
            .setRequiredNetworkType(NetworkType.CONNECTED)
            .build()

        val request = OneTimeWorkRequest.Builder(SendLogWorkerJava::class.java)
            .addTag("sendlog")
            .setConstraints(constraints)
            .build()

        WorkManager.getInstance(context!!).enqueue(request)
    }
}
